### ISSUE TITLE
Specify uri

### ISSUE THREAD

**Issue by wrathematics** [singularityhub/singularityhub.github.io#141]
Is it possible to specify the uri? Basically related to this https://github.com/singularityhub/singularityhub.github.io/issues/15, I want to have a single git repository of recipes, but in my case they will build on each other. Right now, Singularity hub is naming my initial image (on which all others will depend) after the github repo name, "singularity", which isn't what I want https://www.singularity-hub.org/collections/1319. Is there any way to control this?

**Comment by vsoch** [comment]
The tag will be derived from the *extension* of the file called `Singularity` so if your collection is called `RBigData/singularity` you can have a file (anywhere in the repository) called `Singularity.tag1` and the final URI would be `RBigData/singularity:tag1`. If you have dependencies, it's up to you to manage - for example, you might have a continuous integration setup that triggers a build of a second image after some time to allow for a first, or with some check on a commit id, etc.

In summary, you can only have one collection per Github repository. The tags, which you can have as many as you like, are derived from the Singularity file extensions and can be located where you please. We don't do as Docker Hub does and allow for custom specification of file names because it must be the case that no extra metadata is stored. For example, if you disconnected a repository and connected it again, it should build the same without needing to remember to "build this file." etc.

**Comment by vsoch** [comment]
And if you want multiple collection names, you create multiple Github repos. Since this advocates for a modular development style, and a more consistent namespace coinciding with Github (e.g., it's less likely for a user to rename a repository than completely change some file or folder name or organization within it) and better separation of build routines, it's the approach that Singularity Hub takes.

**Comment by wrathematics** [comment]
Oh, I didn't realize the commentary in the issue I referenced was exclusively about tags. So there's no way to create multiple collections with just one github repo?

**Comment by vsoch** [comment]
Nope, afraid not! One collection per repository! Github gives you unlimited public repos, so I hope this isn't an issue for you. If you want to use one collection as a base, then you can just set the Bootstrap From shub, eg:

```bash
Bootstrap: shub
From: RBigData/singularity:tag1
```
This is a decision based on (multiple times) creating single repositories to manage multiple containers (or other static content that I truly believed wouldn't get large). It usually explodes eventually :) If you need to maintain a single folder, then you might consider git submodules, so at least the act of cloning a bunch of recipes (still maintained in separate repos) is a bit more manageable.

**Comment by vsoch** [comment]
And I might add, I hope there's an `RLittleData` too! There is much <3 for the little data, I believe, by robots across the Hub! What would the Big Dipper be without the Little Dipper? ...just a Dipper! :P

**Comment by wrathematics** [comment]
Well my main concern is repo name collisions.

Anyway, question answered. Thanks for the quick response!

**Comment by vsoch** [comment]
Huh, how would there be repo name collisions? Could you give an example? 

Glad to help! Happy building! :hammer: :robot:
